modules/apps/frontend-taglib/frontend-taglib-clay/src/main/resources/META-INF/resources/dataset_display/utilities/odata.js

```diff
@@ -0,0 +1,136 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+export function convertObjectDateToIsoString(objDate, direction) {
+	const time = direction === 'from' ? [0, 0, 0, 0] : [23, 59, 59, 999];
+	const date = new Date(
+		objDate.year,
+		objDate.month - 1,
+		objDate.day,
+		...time
+	);
+
+	return date.toISOString();
+}
+
+function formatStrings(value) {
+	if (typeof value === 'string') {
+		return `'${value}'`;
+	}
+
+	if (Array.isArray(value)) {
+		return value.map((valueEl) => {
+			if (typeof valueEl === 'string') {
+				return `'${valueEl}'`;
+			}
+
+			if (valueEl.value) {
+				return {
+					...valueEl,
+					value:
+						typeof value === 'string'
+							? `'${valueEl.value}'`
+							: valueEl.value,
+				};
+			}
+
+			return valueEl;
+		});
+	}
+
+	return value;
+}
+
+function createOdataFilterString(
```

Comment:
About this and other functions in this file. As a follow-up, we might want to create a ticket to compare this with some other OData manipulation that we are doing elsewhere in liferay-portal ([example from segments-web](https://github.com/liferay/liferay-portal/blob/master/modules/apps/segments/segments-web/src/main/resources/META-INF/resources/js/utils/odata.es.js)). There might be some common low-level utility that we can extract and share.

Reply:
https://issues.liferay.com/browse/LPS-116585

Reply:
🙏